packages/babel-plugin-transform-object-rest-spread/src/index.js

```diff
@@ -21,23 +21,69 @@ export default function({ types: t }) {
     return false;
   }
 
-  function createObjectSpread(file, props, objRef) {
-    const restElement = props.pop();
-
+  // normalizes identifier keys to string literals and extracts all keys
+  // e.g. extracts ["a", "b", 3, ++x] from {a: "foo", b, 3: "bar", [++x]: "baz"}
+  function extractNormalizedKeys(path) {
+    const propPaths = path.get("properties");
     const keys = [];
-    for (const prop of props) {
-      let key = prop.key;
-      if (t.isIdentifier(key) && !prop.computed) {
-        key = t.stringLiteral(prop.key.name);
+    for (const propPath of propPaths) {
+      const key = propPath.get("key").node;
+      if (t.isIdentifier(key) && !propPath.node.computed) {
+        keys.push(t.stringLiteral(key.name));
+      } else {
```

Comment:
I misunderstood earlier.

Are you proposing that we use isLiteral to convert the literals at transform time so we don't have to map to `toPropertyKey`?

It's some preprocessing, but I don't know if it will help the perf since object literals will still have to be coerced to strings before comparison so we can't get rid of `.map(toPropertyKey)` : https://github.com/babel/babel/pull/5757/files/26a3399e17bba349f842351ab0787916523f22a1#diff-7e1831686036bfb55d54af999ed6b91fR13